modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -18,10 +18,10 @@ export class {{classname}} {
     protected basePath = '{{basePath}}';
     public defaultHeaders : Headers = new Headers();
 
-    constructor(protected http: Http, basePath: string) {
```

Comment:
Because of angular2 framework, injector's (this generated class) constructor arguments will be instantiated by the framework (Dependency injection).
To use DI framework provided by angular2, developer have to define [injectable providers name in advance](https://angular.io/docs/ts/latest/cookbook/dependency-injection.html). `string` is not injectable, so when I try to use generated code with previous constructor, it fails with error.